components/script/dom/gpudevice.rs

```diff
@@ -119,6 +148,37 @@ impl GPUDevice {
     pub fn id(&self) -> webgpu::WebGPUDevice {
         self.device
     }
+
+    pub fn handle_server_msg(&self, scope: u64, result: Result<(), GPUError>) {
+        let mut err_scope;
+        {
+            err_scope = self.error_scopes.borrow_mut().remove(&scope).unwrap();
```

Comment:
why are we even trying so eagerly to remove the element from `self.error_scopes`?

Reply:
I figured it was less lines of code if we remove and conditionally reinsert it. Otherwise we first borrow it mutably and then make some checks a second time to see if the scope can be removed.

Reply:
can we use the Entry API there?

Reply:
I have used flags here instead. We set the flags while in-place checking and remove only if the flags are set. Using only Entry wouldn't have made much difference.